LuaRules/Gadgets/game_share_mode.lua

```diff
@@ -67,11 +71,11 @@ end
 
 --------------------------------------------------------
 -- Variables
-local invites = {}
-local controlledPlayers = {} -- table containing which team a playerID should be under.
-local controlledTeams = {} -- contains which team a team of players should be under.
-local originalTeamID = {} -- takes playerID as the key, gives the team as the value.
+local debug = false
+local playerstates = {} -- keeps track of player states so that we can remerge them if necessary.
+local invites = {} -- the invites people have out.
 local originalUnits = {} -- contains which units are owned by a team that has commshared.
+local deadplayers = {count = 0, players = {}}
```

Comment:
Wrapping stuff in a table adds extra indexing
```suggestion
local deadplayers = {}
local deadplayersCount = 0
```

Reply:
That makes it progressively worse, right? (Curiosity)

Reply:
Not progressively but worse nonetheless. Basically if you do `localvar[x]` the generated code is
```
index table localvar with x
```
and if it's `localvar.subtable[x]` it becomes
```
index table localvar with the string "subtable"
index the table from the previous step with x
```
This is also the reason why globals are bad, they're really entries in a global table, so `globalvar[x]`
 is similarly
```
index table _G with the string "globalvar"
index the table from the previous step with x
```

Reply:
Thanks for the explanation. That's really helpful. I'll note this in the future.